client/tracing/src/lib.rs

```diff
@@ -287,3 +295,32 @@ fn send_telemetry(span_datum: SpanDatum) {
 	);
 }
 
+#[derive(Clone)]
+pub struct Metrics {
+	profiling: GaugeVec<U64>
+}
+
+impl fmt::Debug for Metrics {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		write!(f, "Metrics")
+	}
+}
+
+impl Metrics {
+	pub fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+		Ok(Self {
+			profiling: register(GaugeVec::new(
+				Opts::new("profiling", "Profiling from tracing"),
+				&["name", "target", "line", "values"]
+			)?, registry)?,
+		})
+	}
+}
+
+fn send_prometheus(span_datum: SpanDatum, metrics: &Metrics) {
+	metrics.profiling
+		.with_label_values(&[
+			span_datum.name, span_datum.target, &span_datum.line.to_string(), &span_datum.values.to_string()
```

Comment:
Do I understand correctly, that this would add a new metric label combination for each sample? That would result in a [cardinality explosion](https://www.robustperception.io/cardinality-is-key).

What questions would you like to answer with the information exposed here? Keep in mind that Prometheus is a metric system and not a tracing system.

Reply:
Oh, I see. I think we should probably ask @mattrutherford if this is needed.

Reply:
Not entirely sure how this will look on Prometheus side - but might be worth considering if we should collect spans (instead of sending each value) and provide Prometheus histogram instead. I guess it would not be practical to partition things by `values`, but including the line number would be nice. Maybe the metrics names could be derived by combining target + name + line eg: `pallet_society_on_initialize_680` and then have quantiles for that?

Reply:
Thinking about it, I'm not sure there will be much of a cardinality problem given that the name, target and line (and values I think) are tied together.